src/extension/settingsHelper.ts

```diff
@@ -63,4 +67,22 @@ export class SettingsHelper {
                 return settingsJson["typescript.tsdk"] || "";
             });
     }
+
+    /**
+     * Reads and parses the launch.json file.
+     */
+    public static readLaunchJson(): Q.Promise<any> {
+        let fileSystem = new FileSystem();
+        let launchJsonPath = SettingsHelper.launchJsonPath;
+
+        return fileSystem.exists(launchJsonPath)
```

Comment:
Given that for the purpose of this method, the file not being there, or being unable to read it for any reason is the same, I'd argue that we should either return the JSON object, with a resolved promise. And if for any reason we cna't do that, we should just reject the promise and let the caller handle it somehow. Adding a third possible case that we tread identically to the second one, just adds unnecessary complexity.